### PR TITLE
chore(theme): theme POC

### PR DESCRIPTION
**Pull request by carlafranca**
[DS-814](https://jira.equisoft.com/browse/DS-814)

Here is my approach to thematization. 
I would appreciate your input and feedback on it.
Just a reminder this is only a POC.
Lets focus on, if this is a good approach, and if this is flexible enough to help developers to reskin their product.
After we can get the bits and bobs for prod.

**Note:**
I created a folder with the files used to custom the theme.
1. Using a general name (custom-palette) as it can be used by any client
2. The only place to be updated is the custom-theme/custom-palette.ts file with full or partial colour/property, and/or Tokens if need a more granular change.
3. The merge-palette.ts has the functionality to generate a merged pallet checking if the custom property exist in the equisoft-palette and if colour differ it will update it otherwise will keep using the equisoft-palette color.
4. At this point I left the 2 pallets working as I don't know if there is any case scenario where this can be used, but my suggestion is, once you have a custom-pallet in place we should only have merged-palette available and the developer can also update the tokens if the colour-palette is not granular enough.

**Things that I still need to do:**
Map the old names to the new colours name created by Marc-André
Fix the way the theme prop is dealing with the value added as I can see it is adding the style props there after redering.

### CODE REVIEW

File: packages/react/src/themes/custom-theme/merge-palettes.ts
Comment (savutsang) [inline packages/react/src/themes/custom-theme/merge-palettes.ts:1]: If I understand correctly, the user need to write to function to merge his theme with the default.
https://github.com/kronostechnologies/design-elements/blob/2867e59cf9c38e57abb2b36a702f82149bc19c59/packages/react/src/themes/custom-theme/merge-palettes.ts

I think that should be our responsibility to do that, a least provide a function for a standard merge, and if they don't like it, they could do their own custom merge then.
```
const customTheme = createTheme({
  main: {...}
})
```

Comment (carlafranca) [reply packages/react/src/themes/custom-theme/merge-palettes.ts:1]: nope, the only thing you need to do is to update custom-palette.ts to get it all working.

Also, use the new theme where you want, as we should all have a conversation about it.
```
<DesignSystem **theme={customTheme}**>
      <div>
          <h3>Custom theme</h3>
          <Button label="Primary" buttonType="primary" />
          <Button label="Secondary" buttonType="secondary" />
          <Button label="Tertiary" buttonType="tertiary" />
      </div>
  </DesignSystem>
  <DesignSystem>
      <div>
          <h3>Default Equisoft theme</h3>
          <Button label="Primary" buttonType="primary" />
          <Button label="Secondary" buttonType="secondary" />
          <Button label="Tertiary" buttonType="tertiary" />
      </div>
 </DesignSystem>
```

**Review by pylafleur — COMMENTED** [review]
I'm not sure I understand correctly when you say only `custom-palette.ts` will need to be updated. Is your idea to add all custom themes inside the DS?

Also, I think I would merge the themes "within" the DS instead, meaning that I would call the `mergeThemes` function from the `theme-wrapper` or `design-system` files.

Also, I understand this is only a POC but there is no way to customize token values outside of color names. We would need a way to assign different values from the palette to tokens.

**Comment by carlafranca** [comment]
@pylafleur, 
1. You can update custom-palette.ts and tokens.ts if needed.
2. Not sure if I understand this part
3. yes you can use nay value on the tokens, if you want to use any of the main, greys, notification but not only you can also in tokens do as below, is that what you mean?
be aware that if you want to make a change of colour to only one specific button, this can only be done via using classes in the components.
```
'button-primary': {
            bg: "#000000", //instead of `${main['brand-50']}`,
```

Note:
1. I was asked to restrict this to colours only;
2. Also as I was asked to not add classes to Accordion component I guess this makes the customization restricted to the colours as in custom-pallets.ts or to the tokens.ts only.

Beyond this the only way is to add classes to the components.
I did the implementation in the buttons, you will see if you change the colours to any other colour it will work.

Maybe we should all have a call.

**Comment by savutsang** [comment]
Maybe we understand it wrongly, so custom-theme is not the user theme, it's the produced theme from the merge of "user + default" themes.

File: packages/react/src/themes/custom-theme/merge-palettes.ts
Comment (savutsang) [reply packages/react/src/themes/custom-theme/merge-palettes.ts:1]: So "custom-theme" is just an example of an user theme, right? (that in reality all the 4 files would be in the user project).

**Comment by carlafranca** [comment]
created a new repo/PR as there were way too many files to update.

**Pull request closed without merging**